scripts/globals/magic.lua

```diff
@@ -347,8 +347,15 @@ end
 
 function getMagicHitRate(caster, target, skillType, element, percentBonus, bonusAcc)
     -- resist everything if magic shield is active
-    if (target:hasStatusEffect(tpz.effect.MAGIC_SHIELD, 0)) then
```

Comment:
We have 2 versions of this overloaded function:
 - https://github.com/project-topaz/topaz/blob/45c97efb1fdb128570a223b3bea8cb40ed281182/src/map/status_effect_container.cpp#L1110
```cpp
bool CStatusEffectContainer::HasStatusEffect(EFFECT StatusID, uint16 SubID)
{
    for (CStatusEffect* PStatusEffect : m_StatusEffectSet)
    {
        if (PStatusEffect->GetStatusID() == StatusID &&
            PStatusEffect->GetSubID() == SubID &&
            !PStatusEffect->deleted)
        {
            return true;
        }
    }
    return false;
}
```

- https://github.com/project-topaz/topaz/blob/45c97efb1fdb128570a223b3bea8cb40ed281182/src/map/status_effect_container.cpp#L835
```cpp
bool CStatusEffectContainer::HasStatusEffect(EFFECT StatusID)
{
    for (CStatusEffect* PStatusEffect : m_StatusEffectSet)
    {
        if (PStatusEffect->GetStatusID() == StatusID &&
            !PStatusEffect->deleted)
        {
            return true;
        }
    }
    return false;
}
```

Reply:
Ah, that makes much more sense now, I think I'm following your line of reasoning. I'll take a look at applying a "isfakemagicshield" sub for Fanatic's Tonic and Spiritual Incense since these offer the MDT UMDT adjustments, but not the actual magic immunity.